src/measure_theory/integral/periodic.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2021 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import measure_theory.group.fundamental_domain
+import measure_theory.integral.interval_integral
+
+/-!
+# Integrals of periodic functions
+
+In this file we prove that `∫ x in b..b + a, f x = ∫ x in c..c + a, f x` for any (not necessarily
+measurable) function periodic function with period `a`.
+-/
+
+open set function measure_theory measure_theory.measure topological_space
+open_locale measure_theory
+
+lemma is_add_fundamental_domain_Ioc {a : ℝ} (ha : 0 < a) (b : ℝ) (μ : measure ℝ . volume_tac) :
+  is_add_fundamental_domain (add_subgroup.zmultiples a) (Ioc b (b + a)) μ :=
+begin
+  refine is_add_fundamental_domain.mk' measurable_set_Ioc (λ x, _),
+  have : bijective (cod_restrict (λ n : ℤ, n • a) (add_subgroup.zmultiples a) _),
+    from (equiv.of_injective (λ n : ℤ, n • a) (zsmul_strict_mono_left ha).injective).bijective,
+  refine this.exists_unique_iff.2 _,
+  simpa only [add_comm x] using exists_unique_add_zsmul_mem_Ioc ha x b
+end
+
+variables {E : Type*} [normed_group E] [normed_space ℝ E] [measurable_space E] [borel_space E]
+  [complete_space E] [second_countable_topology E]
+
+namespace function
+
+namespace periodic
+
+/-- An auxiliary lemma for a more general `function.periodic.interval_integral_add_eq`. -/
+lemma interval_integral_add_eq_of_pos {f : ℝ → E} {a : ℝ} (hf : periodic f a)
+  (ha : 0 < a) (b c : ℝ) : ∫ x in b..b + a, f x = ∫ x in c..c + a, f x :=
+begin
+  haveI : encodable (add_subgroup.zmultiples a) := (countable_range _).to_encodable,
```

Comment:
Should this be a global instance ?

Reply:
This is noncomputable. I'm going to turn `countable` into a `Prop`-version of `encodable`. Then we can turn it into an instance.